src/NuGet.Core/NuGet.Packaging/Signing/Package/ISignedPackageReader.cs

```diff
@@ -39,10 +39,14 @@ public interface ISignedPackageReader : IDisposable
 
         /// <summary>
         /// Get the hash of the package content excluding signature context for signed package.
+        /// If the package is not signed it calculates it from the whole package.
         /// </summary>
-        /// <param name="token">Cancellation token</param>
-        /// <returns>hash of the unsigned content of the signed package. but return null for unsigned package.</returns>
-        string GetContentHashForSignedPackage(CancellationToken token);
+        /// <param name="token">Cancellation token.</param>
+        /// <param name="fallbackHashGenerator">Function to return the hash in case the package is not signed.
+        /// This is helpfull when the hashFile has already been created and we want to avoid reading that file or calculating the hash again.</param>
+        /// <remarks>The method takes an optional fallback hash filepath to read the hash of an unsigned package instead of calculating it.</remarks>
```

Comment:
I don't see any provision to pass filePath into this fallbackHashGenerator func… am I missing something?